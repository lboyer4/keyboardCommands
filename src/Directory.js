import React, { Component } from 'react';
import './Directory.scss';
