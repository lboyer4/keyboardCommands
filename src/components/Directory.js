import React, { Component } from 'react';
import '../SCSS/_Directory.scss';

class Directory extends Component {
	constructor() {
		super()
		this.state= {
		}
	}

clearGame = () => {
	// e.preventDefault()
	this.props.clearAll()
}

returnStart = () => {
	// e.preventDefault()
	this.props.returnStartPage()
}

	render() {
		return (
				<div className="header"> 
					<button className="restart" onClick={this.returnStart}>Return Home</button>
						<h5 className="unicorn-count">Unicorns Helped: {this.props.score} </h5>
					<button className="clearAll" onClick={this.clearGame}>Reset Game</button>
				</div>

		)
	}
}

export default Directory;