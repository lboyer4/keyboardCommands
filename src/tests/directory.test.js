import React from 'react';
import ReactDOM from 'react-dom';
import Directory from '../components/Directory';
import { shallow } from 'enzyme'; 

describe("Directory", () => {
	let wrapper;

	const mockClearGame = jest.fn();
	const mockClearAll = jest.fn();
	const mockReturnStatePage = jest.fn();

	beforeEach(() => {
		wrapper = shallow(
			<Directory 
				score={0} 
        clearAll={mockClearAll} 
        returnStartPage={mockReturnStatePage} 
        incorrectCards={[]}
			/>
		)
	});

	it('should match snap shot', () => {
		expect(wrapper).toMatchSnapshot();
	});

	 it('should clear all when button is pushed', () => {
    wrapper.find('.clearAll').simulate('click');
    expect(mockClearAll).toBeCalled();
  });

	 it('should return to start when clicked', () => {
	 	wrapper.find('.restart').simulate('click');
	 	expect(mockReturnStatePage).toBeCalled();
	 })
});