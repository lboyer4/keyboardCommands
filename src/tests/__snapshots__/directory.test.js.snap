// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Directory should match snap shot 1`] = `
<div
  className="header"
>
  <button
    className="restart"
    onClick={[Function]}
  >
    Return Home
  </button>
  <h5
    className="unicorn-count"
  >
    Unicorns Helped: 
    0
     
  </h5>
  <button
    className="clearAll"
    onClick={[Function]}
  >
    Reset Game
  </button>
</div>
`;
